fix(material/chips): add opt-out for single-select checkmarks

zarend opened the pull request:
Add an opt-out for checkmark indicators for single-selection. Add both and Input and DI token to specify if checkmark indicators are hidden for single-select. By default display checkmark indicators for single-selection. If both DI token and Input are specified, the Input wins.

PR #25890 adds checkmork indicator for single selection. Add an opt-out to provide a way to have same appearance as before #25890.

Does not affect multiple-selection.

Does not affect behavior when avatar is provided. When avatar is provided, display checkmark indicator when selected. This is the same behavior as before #25890.

API Changes
 - Add `@Input hideSingleSelectionIndicator` to specify if checkmark indicator is displayed for single-selection
 - Add `hideSingleSelectionIndicator` property to `MatChipsDefaultOptions`, which specifies default value for `hideSingleSelectionIndicator`.

crisbeto commented on src/material/chips/chip-listbox.ts:
```diff
@@ -158,6 +158,18 @@ export class MatChipListbox
+    this._hideSingleSelectionIndicator = coerceBooleanProperty(value);
+    this._syncListboxProperties();
+  }
+  protected _hideSingleSelectionIndicator: boolean =
```
This one should be private since we have access to it through the getter already.

crisbeto commented on src/material/chips/chip-option.spec.ts:
```diff
@@ -294,6 +304,20 @@ describe('MDC-based Option Chips', () => {
       });
+
+      it('should display checkmark graphic by default', () => {
+        expect(
```
I think that we should remove this assertion since it's basically just checking that the DI system behaves as expected. The two below that check against the DOM are enough.

crisbeto commented on src/material/chips/chip-option.spec.ts:
```diff
@@ -331,6 +355,64 @@ describe('MDC-based Option Chips', () => {
       });
+
+      it('should display checkmark graphic by default', () => {
+        expect(
```
Same note as above: this assertion can be removed since it's already covered by the DOM assertions.

crisbeto commented on src/material/chips/chip-set.ts:
```diff
@@ -75,6 +77,9 @@ export class MatChipSet
   protected _defaultRole = 'presentation';
 
+  /** Default chip options. */
+  protected _defaultOptions = inject(MAT_CHIPS_DEFAULT_OPTIONS, {optional: true});
```
This doesn't look like it's used anywhere in the chip set. It might be better to move it to the listbox so it's closer to where it's used. We can always move it back here if we start needing it.

zarend replied:
Done I moved all the `_defaultOptions` in this PR so that they're only accessible in the classes that use them.

crisbeto commented on src/material/chips/chip.ts:
```diff
@@ -210,6 +217,9 @@ export class MatChip
   protected basicChipAttrName = 'mat-basic-chip';
 
+  /** Default chip options. */
+  protected _defaultOptions = inject(MAT_CHIPS_DEFAULT_OPTIONS, {optional: true});
```
This is only being used in the chip option so it should be moved there.

zarend replied:
Done I moved all the `_defaultOptions` in this PR so that they're only accessible in the classes that use them.

crisbeto reviewed (approved):
LGTM with a few nits.

zarend commented:
Changes since this was last reviewed
 - moved all the `_defaultOptions` in this PR so that they're only accessible in the classes that use them.
 - made `_hideSingleSelectionIndicator` private
 - removed unnecessary assertions in tests
